### PR TITLE
Refactor IPC communication

### PR DESCRIPTION
**Pull request by donce**

### CODE REVIEW

File: src/renderer/pages/Terms.vue [hunk 1]
```diff
@@ -14,19 +14,21 @@
 </template>
 <script>
   import {mapGetters} from 'vuex'
-  import communication from '../../app/communication'
-  import RendererMessageBus from '../../app/communication/rendererMessageBus'
+  import dependencies from '../dependencies'
```
Comment (Waldz) [inline src/renderer/pages/Terms.vue]: You are running bootstraping DI 2nd time. See example in `CountrySelect.js`

Comment (donce) [reply src/renderer/pages/Terms.vue]: Fixed.

File: src/app/mysterion.js [hunk 2]
```diff
@@ -165,7 +165,7 @@ class Mysterion {
   async startProcess () {
     const updateRendererWithHealth = () => {
       try {
-        this.messageBus.send(busMessages.HEALTHCHECK, this.monitoring.isRunning())
+        this.communication.sendHealthCheck({ status: this.monitoring.isRunning() })
```
Comment (ignasbernotas) [inline src/app/mysterion.js]: Why do we need the status field? Maybe we could just send a boolean?

File: test/helpers/utils.js [hunk 3]
```diff
@@ -20,10 +20,38 @@ async function capturePromiseError (promise: Promise<any>): ?Error {
...
  */
 async function captureAsyncError (func: () => Promise<any>) {
   return capturePromiseError(func())
 }
 
-export { nextTick, capturePromiseError, captureAsyncError }
+/**
+ * Records callback invocation. Useful for asserting that callback was invoked.
+ */
+class CallbackRecorder {
```
Comment (ignasbernotas) [inline test/helpers/utils.js]: Love this. Good idea!

File: src/app/communication/dto.js [hunk 4]
```diff
@@ -16,8 +16,26 @@ export type CurrentIdentityChangeDTO = {
...
+export type RequestConnectionDTO = {
   providerId: string
 }
+
+export type RequestTermsDTO = {
+  content: string
+}
+
+export type TermsAnsweredDTO = {
+  answer: boolean
```
Comment (shroomist) [inline src/app/communication/dto.js]: I suggest `isAccepted` or just `accepted`

File: src/app/communication/dto.js [hunk 5]
```diff
@@ -16,8 +16,26 @@ export type CurrentIdentityChangeDTO = {
...
-export type ProposalUpdateDto = Array<ProposalDTO>
+export type ProposalUpdateDTO = Array<ProposalDTO>
 
-export type RequestConnectionDto = {
+export type RequestConnectionDTO = {
   providerId: string
 }
+
+export type RequestTermsDTO = {
+  content: string
```
Comment (shroomist) [inline src/app/communication/dto.js]: you can name this `html` `htmlContent` to be more descriptive

File: src/app/communication/dto.js [hunk 6]
```diff
@@ -16,8 +16,26 @@ export type CurrentIdentityChangeDTO = {
...
+}
+
+export type AppErrorDTO = {
+  message: string,
+  hint: string,
+  fatal: boolean
+}
+
+export type HealthCheckDTO = {
+  status: boolean
```
Comment (shroomist) [inline src/app/communication/dto.js]: perhaps `isRunning` to be more specific

File: test/helpers/utils.js [hunk 7]
```diff
@@ -20,10 +20,38 @@ async function capturePromiseError (promise: Promise<any>): ?Error {
...
+  }
+
+  /**
+   * Returns function, which records it's invocation and argument
+   * into current instance.
+   *
+   * @returns Function
+   */
+  getCallback (): (any) => void {
+    return this._record.bind(this)
```
Comment (shroomist) [inline test/helpers/utils.js]: nice

File: test/helpers/utils.js [hunk 8]
```diff
@@ -20,10 +20,38 @@ async function capturePromiseError (promise: Promise<any>): ?Error {
...
+  }
+
+  _record (data: any): void {
```
Comment (shroomist) [inline test/helpers/utils.js]: you can do 
`_record(...arguments): void {
this.arguments = arguments
}`
arguments is a bit more clear, and we'd be able to test callbacks invocations with more that 1 parameter

Comment (donce) [reply test/helpers/utils.js]: Renamed this to `argument`.
As for arguments array - it's not good to implement something what is more complex when needed:

https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it
https://en.wikipedia.org/wiki/KISS_principle

**Comment by donce** [comment]
@shroomist fixed comments

**Pull request merged**